### DISCUSSION TITLE
Switch between source/implementation and header

### DISCUSSION THREAD

**Discussion by O136** [helix-editor/helix#7647] [Ideas]
I guess there's no way at the moment with version 23.05 to switch between a `.c(pp)` and `.h` file with key press.
Will be this possible with the next version or it will come later when the plugin feature is done?

Currently it's tedious to jump between them, especially if I have a deep directory structure, I have to use `:open` and write long paths every time, or best case they've been recently accessed and I can use `ga`.

**Comment by pascalkuthe** [comment]
there is no one-to-one association between headers and source files. Even if there was helix does not and will never contain language specific code.

However you can use `gd` to go to the header definition. And in general the file picked is a lot more convenient for opening files than `:open`. If you find yourself constantly switching between a few files `<space>b` or even a split would work even better
